distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/ScheduledJobManager.java

```diff
@@ -20,9 +34,53 @@
 
     private static final Logger log = LoggerFactory.getLogger(ScheduledJobManager.class);
 
+    private final String workerId;
+
+    private final Profiler profiler;
+
+    private final ZkPathsManager paths;
+
+    private final AtomicBoolean isWorkerShutdown;
+    private final DynamicProperty<Long> timeToWaitTermination;
+    private final DynamicProperty<JobDisableConfig> jobDisableConfig;
+
+    private final AtomicProperty<Integer> threadPoolSize;
+    private final ReschedulableScheduler jobReschedulableScheduler;
+
+    /**
+     * Acquires locks for job for workItems, prolongs them and releases
+     */
+    private final PersistentExpiringLockManager lockManager;
+
     //Jobs that currently scheduled. Some of them are running and some will be launched any time.
     private final Map<JobDescriptor, List<ScheduledJobExecution>> scheduledJobs = new ConcurrentHashMap<>();
 
+    public ScheduledJobManager(
+            CuratorFramework curatorFramework,
+            ZkPathsManager paths,
+            Profiler profiler,
+            String workerId,
+            DynamicProperty<DistributedJobManagerSettings> settings,
+            AtomicBoolean isWorkerShutdown
```

Comment:
isWorkerShutdown breaked borders and leaked into ScheduledJobManager. Maybe isWorkerShutdown should stay in worker class, and worker should use and decide whether to call restartJobsAccordingToNewAssignments or not depending on Worker state?

Reply:
I agree. I have a suggestion for `ReschedulableScheduler` API: https://github.com/ru-fix/jfix-stdlib/pull/73

Reply:
fixed